src/dns/coredns.rs

```diff
@@ -138,6 +138,75 @@ impl CoreDns {
+                                let mut reverse_string: String;
+                                let dot_count = ptr_lookup_ip.matches('.').count();
+                                if dot_count >= 31 {
+                                    // its a ipv6
+                                    // parse ip from dns request
+                                    let mut len = 0;
+                                    let mut dots = 0;
+                                    let mut limit = 0;
+                                    for c in ptr_lookup_ip.chars() {
+                                        if dots == 31 {
+                                            break;
+                                        }
+                                        len += 1;
+                                        if c == '.' {
+                                            dots += 1;
+                                        }
+                                    }
+                                    if ptr_lookup_ip.matches('.').count() == 31 {
+                                        limit = 1;
+                                    }
+                                    ptr_lookup_ip = &ptr_lookup_ip[..len-limit];
+                                } else {
+                                    // its a ipv4
+                                    // parse ip from dns request
+                                    let mut len = 0;
+                                    let mut dots = 0;
+                                    let mut limit = 0;
+                                    for c in ptr_lookup_ip.chars() {
+                                        if dots == 4 {
+                                            break;
+                                        }
+                                        len += 1;
+                                        if c == '.' {
+                                            dots += 1;
+                                        }
+                                    }
+                                    if ptr_lookup_ip.matches('.').count() == 4 {
+                                        limit = 1;
+                                    }
+                                    ptr_lookup_ip = &ptr_lookup_ip[..len-limit];
+                                    let ip_octs: Vec<&str> = ptr_lookup_ip.split('.').collect();
+                                    let reverse_ip: Vec<&str> = ip_octs.into_iter().rev().collect();
+                                    reverse_string = reverse_ip.join(".");
+                                    reverse_string = (&reverse_string[1..reverse_string.len()]).to_string();
+                                    ptr_lookup_ip = &reverse_string;
+                                }
+                                // reverse the ip
+                                trace!("perform lookup reverse lookup for ip: {:?}", ptr_lookup_ip.to_owned());
+                if let Ok(lookup_ip) = ptr_lookup_ip.parse() {
```

Comment:
I know the indentation here is whacky but `make validate` on my system blows up if I don't use it

Reply:
@mheon Its because `cargo fmt` is not capable of detecting indentation inside `macros` which we are using for `select`. You can manually indent like https://github.com/containers/aardvark-dns/pull/39 and I'll add rules `clippy` and `fmt` later.

Otherwise indentation really looks off.